Source/WebKit/NetworkProcess/NetworkDataTask.cpp

```diff
@@ -101,6 +101,8 @@ NetworkDataTask::~NetworkDataTask()
 {
     ASSERT(RunLoop::isMain());
     ASSERT(!m_client);
+
+    m_session->unregisterNetworkDataTask(*this);
```

Comment:
It would probably be good to move all the registerNetworkDataTask calls to the NetworkDataTask constructor to maintain symmetry.